app/src/ingest_bem_pdfs.py

```diff
@@ -0,0 +1,172 @@
+
+    logger.info(
+        "Processing PDFs in %s using %s with %s",
+        pdf_file_dir,
+        app_config.embedding_model,
+        doc_attribs,
+    )
+    for file_path in file_list:
+        if not file_path.endswith(".pdf"):
+            continue
+
+        logger.info("Processing file: %s", file_path)
+        with smart_open(file_path, "rb") as file:
+            grouped_texts = _parse_pdf(file)
+
+            doc_attribs["name"] = _get_bem_title(file, file_path)
+            document = Document(content="\n".join(g.text for g in grouped_texts), **doc_attribs)
+            db_session.add(document)
+            chunks = split_into_chunks(document, grouped_texts)
+            for chunk in chunks:
+                _add_chunk(db_session, chunk.content, chunk.document, chunk.tokens)
+
+
+def _parse_pdf(file: BinaryIO) -> list[EnrichedText]:
+    enriched_texts = enrich_texts(file)
+    markdown_texts = add_markdown(enriched_texts)
+    grouped_texts = group_texts(markdown_texts)
+
+    # Assign unique ids to each grouped text before they get split into chunks
+    for text in grouped_texts:
+        text.id = str(uuid.uuid1())
+    assert len(set(text.id for text in grouped_texts)) == len(grouped_texts)
+    return grouped_texts
+
+
+def enrich_texts(file: BinaryIO) -> list[EnrichedText]:
+    "Placeholder function. Will be implemented for DST-414, probably in a different file."
+    outline: list[Heading] = pdf_utils.extract_outline(file)
+    for _heading in outline:
+        pass
+    return []
+
+
+def add_markdown(enriched_texts: list[EnrichedText]) -> list[EnrichedText]:
+    "Placeholder function. Will be implemented for DST-415, probably in a different file."
+    return enriched_texts
+
+
+@dataclass
+class ChunkTemp:
+    "Temporary class as we work through how we want to update document.py:Chunk"
+
+    # Minimal fields from Chunk
+    content: str
+    tokens: int
+    document: Document
+
+    ## Following are new fields
+
+    # For tracability back to a grouped_text
+    grouped_text_id: str  # The id from grouped_texts
+    parano: int
+    text_type: str  # "paragraph" or "list"
+
+    # Flatten 'page' data from grouped_texts
+    page_number: int
+
+    # Flattened 'headings' data from grouped_texts
+    # (heading level is apparent and heading pageno is not needed)
+    headings: list[str]
+    # ['SOME LEVEL 1 HEADING', 'Some Level 2 Heading', 'Oh my! A 3rd level heading']
+
+    # Number of splits (or chunks) the text was split into
+    num_splits: int  # = 1 (if not split)
+    # If not complete (num_splits > 1), specify the index starting from 0
+    split_index: int = 0
+
+
+def split_into_chunks(document: Document, grouped_texts: list[EnrichedText]) -> list[ChunkTemp]:
+    """
+    Given EnrichedTexts, convert the text to chunks and add them to the database.
+    """
+    chunks: list[ChunkTemp] = []
+    for paragraph in grouped_texts:
+        assert paragraph.id is not None
```

Comment:
should we be handling the exception or will this be implemented in another PR?

Reply:
This assertion shouldn't fail unless there's a bug since `id` should be populated at this point in the code. It was added to appease MyPy:
<img width="614" alt="image" src="https://github.com/user-attachments/assets/f8d2430d-415d-49cd-9e1f-0b0a30b3f0dd">